skycatalogs/utils/config_utils.py

```diff
@@ -16,12 +16,81 @@
 CURRENT_SCHEMA_VERSION = '1.2.0'
 
 
+class YamlIncludeLoader(yaml.SafeLoader):
+
+    """YAML Loader that supports file include directives.
+
+    Uses ``!include`` directive in a YAML file to point to another
+    YAML file to be included. The path in the include directive is relative
+    to the top-level file
+
+        storageClasses: !include storageClasses.yaml
+
+    Examples
+    --------
+    >>> with open("document.yaml", "r") as f:
+           data = yaml.load(f, Loader=YamlIncludeLoader)
+
+    Parameters
+    ----------
+    stream :  text io stream
+        The stream to parse.
+
+    This code was adapted from the LSST Science Pipelines Butler.
+    """
+    def __init__(self, filestream):
+        super().__init__(filestream)
+        self._root = os.path.dirname(filestream.name)
```

Comment:
Are `self._root` and `self._current_dir` both needed?   The base class doesn't seem to have a `_root` attribute, so both of these don't seem to be needed.

Reply:
That may be left over from earlier behavior, where all include paths were evaluated relative to initial file. I changed that to be relative to file with the reference.  Will look into it.